_posts/2022-mm-dd-crypto-1/2022-mm-dd-crypto1-history-and-basics.markdown

```diff
@@ -0,0 +1,157 @@
+
+The last requirement can be implemented in one of two ways and usually makes the difference between what's called a "symmetrical" or an "asymmetrical" crypto system. Let's look at the definitions for these two.
+
+#### Symmetrical crypto systems
+A symmetrical crypto system is defined as a system where the same key that is used to encrypt the message also decrypts the cipher. This makes them pretty easy to use and fast, at a price. For example, you can't (or at least shouldn't) share the same key with multiple parties, unless you want to share information with all of them. And you'll most likely have to come up with new keys on a regular basis in a sort of arms race between you and Oscar/Eve. Because they will try to get hold of your current key. And if they ever do, a large portion of your communications will be visible for them.
+
+##### Caesar cipher
+You most probably know this one, maybe by another name. The story goes that [Gaius Julius](https://en.wikipedia.org/wiki/Julius_Caesar) needed to come up with a means of securely getting military orders to his generals. And what he came up with is a solution as simple as smart, considering the resources he reasonably might've had at hand: he shifted the letters in the alphabet. The most commonly known representation of this is that of a rotating disk marked with two alphabets. This disk allows you to quickly shift the letters as needed.
+
+//PLACEHOLDER: image of Caesar cipher disk
+
+The key in this case is the amount of letters you shift. Assuming that `A` can be represented by the number `0`, `B` by the number `1` and so on (you can see where this is going), you can for example use a key of `5` to tell the recipient of your message that you've replaced all `A`s (`0`) with `F`s (`5`), all `B`s with `G`s, etc...
+Side note: you'll see why we start counting at `0` in the example below. It just makes life a little easier.
+
+To decrypt the message, all your recipient has to do is subtract your key from the cipher text, and they'll end up with your original message. Coming back to the notation above, this means that
+> $P$ is a plain text written in the standard latin alphabet (remember the guidelines from the beginning of this post).
+> $C$ is a cipher text, also written in the latin alphabet
+> $K$ is a number that specifies the shift (I'll dive into this in a bit)
+> $e_k$ is shifting the alphabet by a number of steps defined by $K$
+> $d_k$ is shifting the alphabet by a number of steps defined as $-K$ (to invert e_k)
+
+Don't worry, we're almost there. One final problem presents itself. Assume you receive the following cipher text and a key of `5`: `HWDUYTLWFUMD`. Easy enough to do, you convert it into numbers and subtract `5`. But wait, `D` would be represented by `3`... wouldn't that make the original letter `-2`? This is where the final math part for today comes into play. A method called "modulo":
+> The latin alphabet can be represented by the modulo ring of $\mod{26}$
```

Comment:
```suggestion
> The latin alphabet can be represented by the integers modulo $\mod{26}$
```

Otherwise might be confusing if someone finds the concept of modules (which is related to rings). Wiki also calls it "Ring of integers modulo n" https://en.wikipedia.org/wiki/Modular_arithmetic#Integers_modulo_n .